packages/babel-plugin-transform-no-undefined-identifier/src/index.js

```diff
@@ -0,0 +1,27 @@
+export default function() {
+  return {
+    visitor: {
+      CallExpression(path, state) {
+        const args = path.get("arguments");
+        const evaluatedCallee = path.get("callee").evaluate();
+
+        if (!state.file.opts.filename.includes("packages/babel-plugin-")) {
+          return;
+        }
+
+        if (
+          args.length &&
+          args[0].evaluate().value === "undefined" &&
+          evaluatedCallee.deopt.node.object &&
+          evaluatedCallee.deopt.node.object.name === "t" &&
+          evaluatedCallee.deopt.node.property &&
+          evaluatedCallee.deopt.node.property.name === "identifier"
```

Comment:
You clould replace lines 15-18 with `evaluatedCallee.deopt.matchesPattern("t.identifier")`: it is shorter and it also matches `t["identifier"]`.